### PR TITLE
#181414778 feature (Chats): Users should be able to chat on Barefoot Nomad

### PR DESCRIPTION
**Pull request by sndicunguye**
**1. Name of task**

Chat Messages

**2. Pivot Tracker ID**

https://www.pivotaltracker.com/n/projects/2556266/stories/181414778

**3. Branch Name**

ft-chats-181414778

**4. Description of task**

GIVEN the user is online
WHEN they open the chat tab
THEN they should be able to read all of the past chats

GIVEN the user is online
WHEN they open the chat tab
THEN they should be able to post a public chat

GIVEN the user
WHEN they post a chat
THEN it should be able to be echoed to all online users in realtime.

**5. Logic On Actual Implementation**

Create chat messages and retreive all chat messages posted by users logged in Barefoot Nomad

**6. How should it be tested**

- Navigate to `ft-chats-181414778` branch
- `yarn install` to install new dependencies
- Run the server
- Go to `http://localhost:3000/chat` to test this PR

### CODE REVIEW

**Comment by Robertishimwe** [comment]
HOW CAN WE TEST THIS?

**Review by Robertishimwe — APPROVED** [review]
Good job

File: src/database/models/chatmessage.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      ChatMessage.belongsTo(models.User, {
+        foreignKey: 'userId',
+      });
+      ChatMessage.belongsTo(models.Room, {
+        foreignKey: 'userId',
+      });
+    }
+  }
+
+  ChatMessage.init(
```
Comment (didasmbalanya) [inline src/database/models/chatmessage.js:15]: does this automatically add an auto increment ID to this model?

File: src/database/models/user.js [hunk 2]
```diff
@@ -37,6 +37,9 @@ export default (sequelize, DataTypes) => {
       User.hasMany(models.UserAccommodation, {
         foreignKey: 'userId',
       });
+      User.hasMany(models.ChatMessage, {
+        foreignKey: 'userId',
+      });
```
Comment (didasmbalanya) [inline src/database/models/user.js:42]: user and room relationship here?

File: test/chat.spec.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+//         expect(res).to.have.status(201);
+//         expect(res.body).to.have.property('message', 'Message added successfully');
+//       });
+
+//     it('It should get all chat messages ', async () => {
+//         const res = await chai
+//           .request(app)
+//           .post(`/api/chats`)
+//           .set('Authorization', `Bearer ${userOne.token}`)
+//           .send();
```
Comment (didasmbalanya) [inline test/chat.spec.js]: why is this commented?

**Review by didasmbalanya — CHANGES_REQUESTED** [review]
Good work.  just a few things to look into

**Review by Procech20 — COMMENTED** [review]
Please refer to the [PR Template](https://github.com/atlp-rwanda/code-ninjas-be/blob/develop/PR-Template.md) to make the PR description more understandable

**Review by ericus123 — APPROVED** [review]

**Review by amani-eric — APPROVED** [review]

**Pull request merged**